docker/package/package_generator.py

```diff
@@ -2,25 +2,57 @@
 #
 # SPDX-License-Identifier: LicenseRef-MIT-TQ
 
-import os, shutil, sys, subprocess, json
+import os, shutil, sys, subprocess, json, getopt
 from distutils.dir_util import copy_tree
 
 from .model import Package, print_service_file
 from .packages import packages
 
-if sys.argv[1] == "ubuntu":
-    is_ubuntu = True
-elif sys.argv[1] == "fedora":
-    is_ubuntu = False
-else:
-    raise Exception(f"Unexpected package target OS, only 'ubuntu' and 'fedora' are supported.")
+is_ubuntu = None
+is_source = None
+package_to_build = None
 
-if sys.argv[2] == "source":
-    is_source = True
-elif sys.argv[2] == "binary":
-    is_source = False
-else:
-    raise Exception("Unexpected package format, only 'source' and 'binary' are supported.")
+options, _ = getopt.getopt(sys.argv[1:], "", ['os=', 'type=', 'package='])
+for opt, arg in options:
+    if opt == '--os':
+        if arg == "ubuntu":
+            is_ubuntu = True
+        elif arg == "fedora":
+            is_ubuntu = False
+        else:
+            raise Exception(f"Unexpected package target OS, only 'ubuntu' and 'fedora' are supported.")
+    if opt == '--type':
+        if arg == "source":
+            is_source = True
+        elif arg == "binary":
+            is_source = False
+        else:
+            raise Exception("Unexpected package format, only 'source' and 'binary' are supported.")
+    if opt == '--package':
+        package_to_build = arg
+
+if is_ubuntu is None:
+    raise Exception("Target os wasn't specified, use '--os' option to specify it")
+
+if is_source is None:
+    raise Exception("Package format wasn't specified, use '--type' option to specify it")
+
+is_ubuntu = True
+is_source = True
+
+# if sys.argv[1] == "ubuntu":
+#     is_ubuntu = True
+# elif sys.argv[1] == "fedora":
+#     is_ubuntu = False
+# else:
+#     raise Exception(f"Unexpected package target OS, only 'ubuntu' and 'fedora' are supported.")
+
+# if sys.argv[2] == "source":
+#     is_source = True
+# elif sys.argv[2] == "binary":
+#     is_source = False
+# else:
+#     raise Exception("Unexpected package format, only 'source' and 'binary' are supported.")
```

Comment:
I suppose it should be removed from the history.

Reply:
:see_no_evil:

Reply:
Will remove during the final rebase before merging